llms/mlx_lm/lora.py

```diff
@@ -195,7 +196,27 @@ def load_dataset(args):
 
 
 def print_trainable_parameters(model):
-    total_p = sum(v.size for _, v in tree_flatten(model.parameters())) / 10**6
+
+    quantization_bits = 32
+    for _, module in model.named_modules():
+        if isinstance(module, nn.QuantizedLinear):
+            quantization_bits = module.bits
+            break
+
+    all_params = tree_flatten(model.parameters())
+    quantized_params = {
+        k.rsplit(".", 1)[0] for k, _ in all_params if ".biases" in k or ".scales" in k
+    }
+    total_p = 0
+    for k, v in all_params:
+        if "weight" in k:  # skip biases and scales and lora parameters
+            base_name = k.rsplit(".", 1)[0]
+            if base_name not in quantized_params:
+                total_p += v.size
+            else:
+                total_p += v.size * (32 // quantization_bits)
+
+    total_p /= 10**6
```

Comment:
It was not as easy as expected to use our base methods to implement this, but I think you can do something a lot simpler like the following. It's more general as it will get the bits for each model separately and so it's good to use the pattern here in case others copy it.

```suggestion
    def nparams(m):
        if isinstance(m, nn.QuantizedLinear):
            return m.weight.size * (32 // m.bits)
        return sum(v.size for _, v in tree_flatten(m.parameters()))
    leaf_modules = tree_flatten(model.leaf_modules(), is_leaf=lambda m: isinstance(m, nn.Module))
    total_p = sum(nparams(m) for _, m in leaf_modules) / 10**6
```

Reply:
Looks much clearer. Just a quick question, will this include the LoRa parameter as part of the total parameters?

Reply:
I will indeed...and I noticed that difference between the new and old version. I was thinking about whether it makes sense to include them or not.. 

I think it's fine (maybe better) to include, but you could also remove the trainable params from the `total_p` if you wanted to exclude them.

Reply:
@Awni updated the test and did a quick local test, everything looks good.